hive/src/main/java/com/netflix/iceberg/hive/HiveTableOperations.java

```diff
@@ -0,0 +1,224 @@
+                Collections.emptyList(),
+                new HashMap<>(),
+                null,
+                null,
+                ICEBERG_TABLE_TYPE_VALUE);
+      }
+
+      // unnecessary but setting in case tableName.getParameters() starts returning a deep copy.
+      setParameters(newMetadataLocation, tbl);
+
+      if (base != null) {
+        metaStoreClient.alter_table(database, tableName, tbl);
+      } else {
+        metaStoreClient.createTable(tbl);
+      }
+      threw = false;
+    } catch (TException|UnknownHostException e) {
+      throw new RuntimeException(e);
+    } finally {
+      if (threw) {
+        // if anything went wrong, clean up the uncommitted metadata file
+        deleteFile(newMetadataLocation);
+      }
+      unlock(lockId);
+    }
+
+    refresh();
+  }
+
+  private void setParameters(String newMetadataLocation, Table tbl) {
+    Map<String, String> parameters = tbl.getParameters();
+
+    if (parameters == null) {
+      parameters = new HashMap<>();
+    }
+    parameters.put(TABLE_TYPE_PROP, ICEBERG_TABLE_TYPE_VALUE.toUpperCase(Locale.ENGLISH));
+    parameters.put(METADATA_LOCATION_PROP, newMetadataLocation);
+    if (currentMetadataLocation() != null && !currentMetadataLocation().isEmpty()) {
+      parameters.put(PREVIOUS_METADATA_LOCATION_PROP, currentMetadataLocation());
+    }
+
+    tbl.setParameters(parameters);
+  }
+
+  private StorageDescriptor storageDescriptor(Schema schema) {
+    // TODO HardCoding the parquet for now
+    final StorageDescriptor storageDescriptor = new StorageDescriptor();
+    storageDescriptor.setCols(columns(schema));
+    storageDescriptor.setLocation(hiveTableLocation());
+    storageDescriptor.setOutputFormat("org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat");
+    storageDescriptor.setInputFormat("org.apache.hadoop.hive.ql.io.parquet.MapredParquetInputFormat");
+    SerDeInfo serDeInfo = new SerDeInfo();
+    serDeInfo.setSerdeType(SerdeType.HIVE);
+    serDeInfo.setSerializationLib("org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe");
+    storageDescriptor.setSerdeInfo(serDeInfo);
+    return storageDescriptor;
+  }
+
+  private final List<FieldSchema> columns(Schema schema) {
+    return schema.columns().stream().map(col -> new FieldSchema(col.name(), HiveTypeConverter.convert(col.type()), "")).collect(Collectors.toList());
+  }
+
+  private Optional<Long> acquireLock() throws UnknownHostException, TException {
+    final LockComponent lockComponent = new LockComponent(LockType.EXCLUSIVE, LockLevel.TABLE, database);
+    lockComponent.setTablename(tableName);
+    final LockRequest lockRequest = new LockRequest(Lists.newArrayList(lockComponent),
+            System.getProperty("user.name"),
+            InetAddress.getLocalHost().getHostName());
+    LockResponse lockResponse = metaStoreClient.lock(lockRequest);
+    LockState state = lockResponse.getState();
+    Optional<Long> lockId = Optional.of(lockResponse.getLockid());
```

Comment:
Why is this an Optional? Is there a case where the lock ID is null? Maybe this should be an assertion that the ID is not null?

Reply:
It is optional as unlock is called in finally and Thrift call to acquire lock can fail.

Reply:
Right, but there's no way for this method to return an empty Optional, right? A failure to lock would throw an exception that prevents this method from returning an object. And, `lockReponse.getLockId()` should always be non-null. If that's the case, then this should return a `long` and the caller should wrap it in an `Option` to be clear about what can happen in this method.